### PR TITLE
Attempt to upload heap dump if tests failed

### PR DESCRIPTION
**Pull request by nineinchnick**
## Description

Attempt to upload heap dump if tests failed, but only when there are no secrets defined, and running on a pull request from a repository different than `trinodb/trino`.

This is supposed to help debug OOM errors that happen in the CI.

Was partially discussed in #14865

Previous attempt: #15129

## Additional context and related issues

n/a

## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

### CODE REVIEW

**Comment by findepi** [comment]
> Previous attempt: #15129

what's the key diff?

**Comment by nineinchnick** [comment]
Here I revert adding a separate action for uploading test results and reports. It turned out to require too much overhead and didn't remove enough duplication. Uploading heap dumps is exactly the same.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -462,6 +496,8 @@ jobs:
       fail-fast: false
       matrix: ${{ fromJson(needs.build-test-matrix.outputs.matrix) }}
     timeout-minutes: 60
+    env:
+      SECRETS_PRESENT: ${{ secrets.SECRETS_PRESENT }}
```
Comment (hashhar) [inline .github/workflows/ci.yml:500]: why the indirection? Why not inline `secrets.SECRETS_PRESENT`?

Also if you need it everywhere you can add it to top-level `env` key probably.

**Review by hashhar — APPROVED** [review]
LGTM % question/suggestion

File: .github/workflows/ci.yml (hunk 1 above)
Comment (nineinchnick) [reply .github/workflows/ci.yml:500]: Secrets are not available in condition blocks, this is what githubs manual recommends

**Comment by findepi** [comment]
🎉

**Pull request merged**